bittensor/core/extrinsics/commit_reveal.py

```diff
@@ -125,12 +125,18 @@ def commit_reveal_v3_extrinsic(
             netuid=netuid
         )
 
+        tempo = subtensor.get_subnet_hyperparameters(netuid).tempo
+        current_block = subtensor.get_current_block()
+
```

Comment:
```suggestion
        subnet_hyperparameters = subtensor.get_subnet_hyperparameters(netuid)
        
        # get subnet `tempo` hyperparameter
        tempo = subnet_hyperparameters.tempo
        # get subnet `commit_reveal_weights_interval` hyperparameter
        subnet_reveal_period_epochs = subnet_hyperparameters.commit_reveal_weights_interval

        current_block = subtensor.get_current_block()
```

Reply:
Please test this changes with the installed `bittensor-commit-reveal` python package.

Reply:
Also, please consider adding arguments `tempo`, `current_block, `netuid`, and ` subnet_reveal_period_epochs` to `Subtensor.set_weights` - > `commit_reveal_v3_extrinsic` with default values to `None`. 
@thewhaleking could help you with these changes on the Python side if you don't mind.